ExcelWriter.py

```diff
@@ -19,6 +20,36 @@ def __init__(self, name):
             self.maxCol = []
             self.indentFormat = {}
             self.name=name
+            self.writtenCells = {}
+            
+    def Add_Header(self, headerLines):
+        row = 0;
+        maxCol = 0;
+        for header in headerLines:
+            col = 0
+            for headerLine in header:
+                self.Write_String(row, col, headerLine)
+                col = col + 1
+                maxCol = max(col, maxCol)
+        self.currentCol = maxCol
+        
+    
+    def Write_String(self, row, col, stringVal, formatValue):
+        # by putting chr(2) at the front of the string, google docs will not remove
+        # the leading spaces
+        tabString = " " * 4
+        stringVal = stringVal.replace("\t", tabString)
+        self.Inc_Max_Col(stringVal)
+        
+
+        stringVal = chr(2) + stringVal;
+        stringVal = stringVal.strip()
+        print(stringVal)
+        # storing it in col/row because we are going to walk down the columns & merge the cells
+        if (col not in self.writtenCells):
+            self.writtenCells[col] = []       
```

Comment:
I think you could avoid this branch by initially constructing `self.writtenCells` as a `collections.defaultdict(list)`.

Reply:
Uploaded a new version with this change + the cell merging actually works now.